odoo/addons/base/models/ir_qweb.py

```diff
@@ -36,7 +405,77 @@
 ])) - _BLACKLIST
 
 
-class IrQWeb(models.AbstractModel, QWeb):
+# eval to compile generated string python code into binary code, used in `_compile`
+unsafe_eval = eval
+
+
+VOID_ELEMENTS = frozenset([
+    'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'keygen',
+    'link', 'menuitem', 'meta', 'param', 'source', 'track', 'wbr'])
+# Object available globally in functions compiled by the IrQWeb.
+AVAILABLE_OBJECTS = dict(_BUILTINS)
+# Terms allowed in addition to AVAILABLE_OBJECTS when compiling python expressions
+ALLOWED_KEYWORD = ['False', 'None', 'True', 'and', 'as', 'elif', 'else', 'for', 'if', 'in', 'is', 'not', 'or']
```

Comment:
Should be a frozenset with the data of `AVAILABLE_OBJECTS.keys()` (only for in operation)